### ISSUE TITLE
Imposter Improvement

### ISSUE THREAD

**Issue by Hockeyfan360** [TotalFreedom/TotalFreedomMod#362]
This is a very simple idea that could reduce suspensions and rouge ability. I was imposed by on FOP a while back. Another SA logged in under my username, and abused.  They were supered, in-case you hadn't caught on yet. (They eventually they got removed) That was a while back, but it made me think. TFM should dis-allow admins logging in as each other's last log-in name.

For Example:
Another admin tries to impose as me.
(TFM checks if the admin is logged in under any admin's last login name)
If they are trying to impose, they get kicked.
If they are not imposing, TFM lets them through.

If you have any questions, please ask away.

**Comment by JeromSar** [comment]
I think @Wild1145's issues with this implementation pose a problem. Perhaps there are other ways impostor management could be handled.

**Comment by Wild1145** [comment]
@JeromSar Personally I would completely remove the IP part of giving someone SA. I'd use it to match with their username, but wouldnt allow it to grant admin access based just off of the IP. That way you cant admin family members or alt accounts...

**Comment by JeromSar** [comment]
@Wild1145 Forgotten that TF is an offline mode server?

**Comment by Wild1145** [comment]
@JeromSar Not at all. However the way that this would work has been proven on CJFreedom and other development based servers to be far more effective. Currently it is your name OR IP that will give you admin permissions. I say switch it to require name AND IP, that way siblings can login without getting supered and you can still ensure that players should be receiving admin permissions.

**Comment by JeromSar** [comment]
@Wild1145 No currently, only the IP is required to gain admin perms. Having a name that doesn't match the one on file will result in a warning on login.

**Comment by Wild1145** [comment]
@JeromSar Thats my point, make it so it REQUIRES both the name AND IP to match, that way it solves the problem with siblings and admins acting like idiots and logging in as "notch" or something stupid and getting admin access.

**Comment by Wild1145** [comment]
Another shameless bump here. But I think we should look at changing the verification process for TFM to check **BOTH** the users username and IP. Currently as @JeromSar mentioned IP is all you need to verify, but if TFM required you to match both name and IP it would certainly resolve the family admin policies, and the shenanigans as mentioned earlier. The only thing we would have to look at working around would be the fact players can change names, but it may also deter admins from changing their names every 30 days and confusing everyone...

**Comment by JeromSar** [comment]
@Wild1145 I agree, if the username doesn't match any admin, no perms will be given. Perhaps a personal message along the lines of "You are acting undercover. Other admins do not know you are an admin". Lastly, if the username matches an admin username _not_ registered to that IP, the login could be blocked. This prevents admins imposing each other.

We should also think about the sibling program. We could allow admin entries side by side to exist if we match both username and IP.

**Comment by Wild1145** [comment]
@JeromSar It would also resolve issues such as #1783 where two people on the same IP cannot always connect properly.
